### ISSUE TITLE
Warning about grepping for *

### ISSUE THREAD

**Issue by bje-** [koalaman/shellcheck#1216]
This is a tricky one. I understand the need for this warning, but I can't find a way to sensibly `grep` for a real asterisk without having to explicitly silence the warning.

#### For bugs

``Grep uses regex, but this looks like a glob. [SC2063]``
- Shellcheck version 0.4.6
- [X] I tried on shellcheck.net and verified that this is still a problem on the latest commit

#### For new checks and feature suggestions
- [X] shellcheck.net (i.e. the latest commit) currently gives no useful warnings about this
- [X] I searched through https://github.com/koalaman/shellcheck/issues and didn't find anything related

#### Here's a snippet or screenshot that shows the problem:

```sh

#!/bin/sh
grep \* /etc/passwd
```

#### Here's what shellcheck currently says:

```
warning: Grep uses regex, but this looks like a glob. [SC2063]
```

#### Here's what I wanted or expected to see:

No warning.

**Comment by jleffler** [comment]
Formally, you should make sure `grep` sees a backslash and an asterisk: `grep '\*' /etc/passwd`.

You get away with `grep \* /etc/passwd` because `grep` recognizes that `*` on its own can only match `*` because there's nothing for it to repeat zero or more times.

**Comment by bje-** [comment]
Thanks.
